client/update.go

```diff
@@ -93,6 +94,9 @@ type (
 		req     *msgChannelUpdate
 		called  atomic.Bool
 	}
+
+	// PeerNotRespondingError indicates that a peer has not responded within the expected time period.
+	PeerNotRespondingError string
```

Comment:
Im not quite happy with the name, since it sound like the peer is not responding at all.  
But in this case it only did not respond to a specific request (or we did not receive it).  
Maybe something like `RequestTimedOutError` or `NoResponseError`.

Reply:
I think `RequestTimedOutError` looks better. I will use this one.